### PR TITLE
Fix/docker worker

### PR DESCRIPTION
**Pull request by dsludwig**
Working directory is where the executing process should `cd` into
before execution. For the docker worker, this is relative to the
container file system.

Staging directory is where temporary files should be stored while
setting up a build, on the host machine. For the regular worker, this is
the same as the working directory.

Issue #225

### CODE REVIEW

File: dev-builder.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+rm docker/dev-builder/anaconda-build-*
+python setup.py sdist --dist-dir docker/dev-builder/
+VERSION=$(python -c 'import binstar_build_client; print(binstar_build_client.__version__.replace("+", "-"))')
+(cd docker/dev-builder; docker build -t binstar/linux-64:$VERSION -t binstar/linux-64:dev .)
```
Comment (PeterDSteinberg) [inline dev-builder.sh]: @dsludwig I'm kind of new to docker.  Can you explain how this dev-builder.sh file is used?

Comment (dsludwig) [inline dev-builder.sh]: Good question! The idea: generate a docker image that includes the latest (checked-out) `anaconda-build` code. As I'm describing this.. I'm realizing it's not that useful - most of the interesting stuff happens outside the container. I might remove it.

**Comment by PeterDSteinberg** [comment]
@dsludwig What should I do to fully test this PR?

**Comment by dsludwig** [comment]
* register a new worker:
```
anaconda worker register -n docker-builder me/queue
```
* run the worker:
```
anaconda worker docker_run docker-builder
```
* submit a job (I use https://github.com/Anaconda-Server/testci):
```
anaconda build submit testci
```

You shouldn't see any errors like: https://github.com/Anaconda-Server/anaconda-build/issues/225

**Comment by PeterDSteinberg** [comment]
LGTM.  Tested locally.

**Pull request merged**